src/dawn/Serialization/IIRSerializer.cpp

```diff
@@ -526,13 +553,32 @@ void IIRSerializer::deserializeMetaData(std::shared_ptr<iir::StencilInstantiatio
   for(auto globalVariableID : protoMetaData.globalvariableids()) {
     metadata.fieldAccessMetadata_.GlobalVariableAccessIDSet_.insert(globalVariableID);
   }
+  for(auto allocatedFieldID : protoMetaData.allocatedfieldids()) {
+    metadata.fieldAccessMetadata_.AllocatedFieldAccessIDSet_.insert(allocatedFieldID);
+  }
 
   for(auto variableVersionMap : protoMetaData.versionedfields().variableversionmap()) {
     for(auto versionedID : variableVersionMap.second.allids()) {
       metadata.insertFieldVersionIDPair(variableVersionMap.first, versionedID);
     }
   }
 
+  struct DeclStmtFinder : public ast::ASTVisitorForwarding {
```

Comment:
Since it is serializing IIR, here you need to use the mirrored classes in  `iir:*` namespace. See comment above.